app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-ecommerce-tracker.js

```diff
@@ -46,101 +43,29 @@
       }
     },
 
-    trackSearchResults: function (searchResultsBlock) {
-      var schema = this.populateEcommerceSchema(searchResultsBlock, false, null)
+    trackResults: function (searchResultsBlock) {
+      var ecommerceSchema = GOVUK.analyticsGa4.core.ecommerceHelperFunctions.populateEcommerceSchema({
+        element: searchResultsBlock,
+        resultsId: 'js-result-count'
+      })
 
-      this.clearPreviousEcommerceObject()
-      GOVUK.analyticsGa4.core.sendData(schema)
+      GOVUK.analyticsGa4.core.ecommerceHelperFunctions.clearEcommerceObject()
+      GOVUK.analyticsGa4.core.sendData(ecommerceSchema)
```

Comment:
Just to confirm - do we have to clear out the dataLayer before every push?

Reply:
We do - according to the implementation guide, it's considered best practice to clear out the ecommerce object anytime we push a new one to the `dataLayer`